src/hal/src/window.rs

```diff
@@ -200,6 +200,8 @@ pub struct SwapchainConfig {
     pub depth_stencil_format: Option<Format>,
     /// Number of images in the swapchain.
     pub image_count: u32,
+    /// Image usage of the backbuffer images.
+    pub image_usage: Option<image::Usage>,
```

Comment:
why is this optional?

Reply:
You are correct, it actually isn't as "default" values are applied in create_swapchain in vulkan and dx12. What do you think the best way to handle this would be, just setting it to empty and force users to pass in the required flags, instead of setting the current "default" values in create_swapchain?

Reply:
> "default" values are applied in create_swapchain in vulkan and dx12

where do you see the default usage value in Vulkan?

Reply:
By the "default" value I mean the value that was supplied in the Vulkan backend, https://github.com/gfx-rs/gfx/pull/1906/files#diff-055df6f698963e935780372c2266b221L1421 and for the DX12 backend https://github.com/gfx-rs/gfx/pull/1906/files#diff-30c7631df6499e30d85955a6141462e4L2599

Reply:
Let's just require the user to pass in the flags in full, no defaults needed